python/pyarrow/tests/parquet/test_basic.py

```diff
@@ -882,3 +884,141 @@ def test_thrift_size_limits(tempdir):
     assert got == table
     got = pq.read_table(path)
     assert got == table
+
+
+def test_page_checksum_verification_write_table(tempdir):
```

Comment:
Could we add one more example similar to this with `write_to_dataset` (`test_page_checksum_verification_write_to_dataset(tempdir)`)?

The cases to be handled are:
- `pyarrow.parquet.write_to_dataset` to write and `pq.read_table` to read.
- `pyarrow.parquet.write_to_dataset` with `use_legacy_dataset=True` to write and `pq.read_table` to read.

The test will need to use `@pytest.mark.dataset` and the second might also need `@pytest.mark.filterwarnings("ignore:Passing 'use_legacy_dataset=True':FutureWarning")`.

Reply:
Thanks! In the latest push, I added `test_checksum_write_to_dataset()` to `python/pyarrow/tests/test_dataset.py`, which should implement the two test cases you mentioned.